### ISSUE TITLE
nested inputObjectType with same property names

### ISSUE THREAD

**Issue by u9520107** [graphql/graphql-js#206]
I am not sure if this is a bug or I am missing some reading.
Suppose my mutation takes in an argument `person`, of type Person.
Person has a field named `id`, and a field `tools` which is a list of type Tool.
Tool also has a field named `id`.

The problem is, when I indeed serialize the argument object into graphql.
```
mutation M {
  test: testMutation (
    person: {
      id: 'test-id',
      tools: [{
        id: 'test-tool-id'
      }]
    }
  ) {
    result
  }
}
```

The query results in an error saying that there can only be one input field named "id".

Is the format I am using wrong?

**Comment by u9520107** [comment]
Err.. Probably a bug. This is very weird.
In my test code, I first created an Immutable Map to keep track of changes to the Person object. Then I compare each field to the original copy. If changes are found, I then set id to the diff object. So... after serializing the object, id is the last property. I do the same thing for tools.
So the resulting query looks like this:
```
mutation M {
  test: testMutation (
    person: {
      experience: 123,
      tools: [{
        used: 3,
        id: 'test-tool-id'
      }],
      id: 'test-id'
    }
  ) {
    result
  }
}
```

Then the query would end up in error saying that there can only be one input field named 'id".
If I assign ids to the diff object first, hence id field is on top. Then the query passes...
